includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -120,4 +138,39 @@ public function capture_terminal_payment( $request ) {
 			return new WP_Error( 'wcpay_server_error', __( 'Unexpected server error', 'woocommerce-payments' ), [ 'status' => 500 ] );
 		}
 	}
+
+	/**
+	 * Returns customer id from order. Create or update customer if needed.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function create_customer( $request ) {
+		try {
+			$order_id = $request['order_id'];
+
+			// Do not process non-existing orders.
+			$order = wc_get_order( $order_id );
+			if ( false === $order ) {
+				return new WP_Error( 'wcpay_missing_order', __( 'Order not found', 'woocommerce-payments' ), [ 'status' => 404 ] );
+			}
+
+			$order_user    = $order->get_user();
+			$customer_data = WC_Payments_Customer_Service::map_customer_data( $order );
+			if ( ! $order_user ) {
+				$customer_id = $this->customer_service->create_customer_for_user( new WP_User(), $customer_data );
+			} else {
+				$customer_id = $this->customer_service->get_customer_id_by_user_id( $order_user->ID );
+				if ( $customer_id ) {
+					$customer_id = $this->customer_service->update_customer_for_user( $customer_id, $order_user, $customer_data );
+				} else {
+					$customer_id = $this->customer_service->create_customer_for_user( $order_user, $customer_data );
+				}
+			}
+
+			return $customer_id;
+		} catch ( \Throwable $e ) {
+			Logger::error( 'Failed to create / update customer from order via REST API: ' . $e );
```

Comment:
I think `$e` is an object, you might need `$e->getMessage()`

Reply:
Good catch. I shamelessly copy-pasted that part from the function above it, `capture_terminal_payment()`.
I think it's OK to do so though because [Throwable's __toString()](https://www.php.net/manual/en/class.throwable.php) will be used to get its string representation.

I tried this:
```php
$e = new Exception('abc');
$a = '$e: ' . $e;
var_dump($a);
```
and it outputs not just the message but also the stack:
```
string(65) "$e: Exception: abc in /private/tmp/a.php:2
Stack trace:
#0 {main}"
```

Reply:
> I think it's OK to do so though because Throwable's __toString() will be used to get its string representation.
Neat, I didn't know about Throwable's `__toString()`